### ISSUE TITLE
Is the Model Capable of Processing and Maintaining Consistent Output Sizes Across Varied Image Dimensions?

### ISSUE THREAD

**Issue by arrrrr3186** [junyanz/pytorch-CycleGAN-and-pix2pix#1633]
@junyanz @ssnl @AyushExel 

So far i know, model can accept the any image size then implement the preprocess step. 

can Model handle images of different sizes and generate outputs in the same size? For example, if I throw in a pic that's 256x256 and another that's 500x300, can the model keep the output sizes consistent?

Curious to know! Thanks a bunch!

**Comment by arrrrr3186** [comment]
Hi there, anyone can help with this problem, 
it would be a great help to me.

**Comment by ssnl** [comment]
You can resize the input to the same size before feeding into the model.

**Comment by arrrrr3186** [comment]
Thank you for response.

Simply resizing the input image won't address my issue. Let's say I resize an image to a standard size like 800x800 pixels. But what if your original input size is larger, like 1024x1024 pixels? In that case, resizing would change the dimensions to 800x800 pixels.

What I'm looking for is consistency in output size, regardless of the input image's dimensions. Whether it's 800x800, 1024x1024, or even 256x256 pixels, I want the output image to maintain the same dimensions as the input.

Do you know if there's a way to achieve this?

**Comment by ssnl** [comment]
What’s the problem with resizing to 800x800?

**Comment by arrrrr3186** [comment]
Hi,

the problem with this is that, If I pass input image size 1024*1024, then the output will be 800*800.
but I want the output to be 1024*1024.

**Comment by ssnl** [comment]
Oh that's what you mean. It was ambiguous from your description. 

The model is convolutional so it can probably take in any reasonably sized image and spit out a similarly sized (exact size would depend on strides and padding). However, the pretrained model are not trained on other sizes so no performance guarantees. 

You can modify the code to train new models on arbitrarily sized data. We make no promises on how well it works.
